midend/simplifyBitwise.cpp

```diff
@@ -0,0 +1,98 @@
+#include "simplifyBitwise.h"
+
+namespace P4 {
+
+bool SimplifyBitwise::Scan::preorder(const IR::AssignmentStatement *as) {
+    if (!as->right->is<IR::BOr>())
+        return false;
+    can_be_changed = true;
+    mpz_init(total_mask.get_mpz_t());
+    return true;
+}
+
+bool SimplifyBitwise::Scan::preorder(const IR::BOr *bor) {
+    if (!findContext<IR::AssignmentStatement>())
+        can_be_changed = false;
+    if (!(bor->left->is<IR::BAnd>() && bor->right->is<IR::BAnd>()))
+        can_be_changed = false;
+    return can_be_changed;
+}
+
+bool SimplifyBitwise::Scan::preorder(const IR::BAnd *band) {
+    if (!findContext<IR::BOr>())
+        can_be_changed = false;
+    if (!band->left->is<IR::Constant>() && !band->right->is<IR::Constant>())
+        can_be_changed = false;
+    if (band->left->is<IR::Constant>() && band->right->is<IR::Constant>())
+        can_be_changed = false;
+    return can_be_changed;
+}
+
+bool SimplifyBitwise::Scan::preorder(const IR::Constant *constant) {
+    if (!findContext<IR::BAnd>()) {
+        can_be_changed = false;
+        return false;
+    }
+    // Ensure that there are no collisions within the bitmask
+    mpz_class mask;
+    mpz_init(mask.get_mpz_t());
+    mpz_class zero;
+    mpz_init(zero.get_mpz_t());
+    mpz_and(mask.get_mpz_t(), total_mask.get_mpz_t(), constant->value.get_mpz_t());
```

Comment:
The above should be

    mpz_class mask = total_mask & constant->value;

basically you should just be using `mpz_class` objects as if they were integers (and can intermix them with `int` and `long`) and they "just work"